src/integration-test/integration/MappingsClient.test.ts

```diff
@@ -0,0 +1,485 @@
+*--------------------------------------------------------------------------------------------*/
+import * as chaiAsPromised from "chai-as-promised"
+import { expect, use } from "chai";
+import { MappingsClient, MappingCreate, GroupCreate, GroupPropertyCreate, CalculatedPropertyCreate, CustomCalculationCreate, MappingUpdate, ECProperty, GroupUpdate, CalculatedPropertyUpdate, CustomCalculationUpdate, GroupPropertyUpdate, MappingCopy, DataType, QuantityType, CalculatedPropertyType } from "../../reporting";
+import "reflect-metadata";
+import { AuthorizationCallback, getTestRunId, TestConstants, getTestDIContainer, TestIModelGroup, TestIModelGroupFactory, IModelMetadata, TestAuthorizationProvider, TestIModelCreator, ReusableTestIModelProvider } from "../utils/";
+use(chaiAsPromised);
+
+describe("Mapping Client", () => {
+  const mappingsClient: MappingsClient = new MappingsClient();
+  let accessToken: string;
+
+  const mappingIds: Array<string> = [];
+  let groupId: string;
+  let groupPropertyId: string;
+  let calculatedPropertyId: string;
+  let customCalculationId: string;
+
+  let authorization: AuthorizationCallback;
+  let testIModelGroup: TestIModelGroup;
+  let testIModel: IModelMetadata;
+
+  before( async function () {
+    this.timeout(0);
+
+    const container = getTestDIContainer();
+    
+    const authorizationProvider = container.get(TestAuthorizationProvider);
+    authorization = authorizationProvider.getAdmin1Authorization();
+    accessToken = "Bearer " + (await authorization()).token;
+
+    const testIModelGroupFactory = container.get(TestIModelGroupFactory);
+    testIModelGroup = testIModelGroupFactory.create({ testRunId: getTestRunId(), packageName: TestConstants.PackagePrefix, testSuiteName: "ManagementNamedVersionOperations" });
+
+    const testIModelCreator = container.get(TestIModelCreator);
+    testIModel = await testIModelCreator.createEmptyAndUploadChangesets(testIModelGroup.getPrefixedUniqueIModelName("Test iModel for write"));
+
+    const reusableTestIModelProvider = container.get(ReusableTestIModelProvider);
+    testIModel = await reusableTestIModelProvider.getOrCreate();
+
+    //create mappings
+
+    const newMapping: MappingCreate = {
+      mappingName: "Test1"
+    };
+    let mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    mappingIds.push(mapping.id);
+
+    newMapping.mappingName = "Test2";
+    mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    mappingIds.push(mapping.id);
+
+    newMapping.mappingName = "Test3"
+    mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    mappingIds.push(mapping.id);
+
+    //create groups
+
+    const newGroup: GroupCreate = {
+      groupName: "Test1",
+      query: "select * from biscore.element"
```

Comment:
Add `LIMIT` clauses to all Group queries that you create so that we do not run excessively long extractions for testing. I think we can limit it to 10 rows to be safe.

Reply:
added limits to all queries